src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,34 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (checkUser(user)) {
+            storageDao.add(user);
+        }
+        if (user.getLogin() == null
+                || user.getPassword() == null
+                || user.getId() == null
+                || user.getAge() == null) {
+            throw new ValidationException("Invalid data, check all user values");
+        }
+        return user;
+    }
+
+    private boolean checkUser(User user) {
+        if ((storageDao.get(user.getLogin()) == null)
+                && (user.getAge() >= MIN_AGE)
+                && (user.getPassword().length() >= MIN_PASSWORD_LENGTH)) {
+            return true;
+        }
+        return false;
```

Comment:
You might (or might not - I don't know if it is required) create different checks for different fields with different error messages, e.g. null login or password. I believe it's optional though, so it's up to you wheather to implement it or not

Reply:
I added a general message if a field does not comply with the requirements, but if I add a message for each check, it would have to change the entire structure of the entire class, because in this case the boolean method will be useless and everything will need to be prescribed in the register method, because I can't add to each check error and return True or False

Reply:
I don't know if it is required, probably not, so for me it's ok if you don't want to re-write the entire structure of your class